### PR TITLE
Return 0-numel empty tensor from symeig when eigenvectors=False

### PR DESCRIPTION
**Pull request by vishwakftw**
Changelog:
- Changes the behavior of returning a zero tensor when eigenvectors=False, matching behavior of torch.eig

Test Plan:
- test_symeig has been modified appropriately for this change

### CODE REVIEW

**Comment by vishwakftw** [comment]
If this change is fine, I'll send a PR to pytorch/xla to fix the XLA error.

File: test/test_torch.py [hunk 1]
```diff
@@ -8433,7 +8433,7 @@ def run_test(dims, eigenvectors, upper):
             else:
                 eigvals, _ = torch.symeig(x, eigenvectors=True, upper=upper)
                 self.assertEqual(eigvals, rese, 'Eigenvalues mismatch')
-                self.assertEqual(torch.zeros_like(resv), resv, 'Eigenvector matrix not zero')
+                self.assertEqual(torch.empty(0, device=device, dtype=dtype), resv, 'Eigenvector matrix not zero')
```
Comment (Balandat) [inline test/test_torch.py]: zero -> empty

**Review by Balandat — APPROVED** [review]
These changes look good to me. Presumably no one is using the zero tensors if they call `symeig` with `eigenvectors=False` - if they do, this change may even help them uncover bugs...

**Comment by vishwakftw** [comment]
cc: @ailzhang who originally introduced zeroing and also regarding XLA failure

**Review by ailzhang — APPROVED** [review]
Thanks! 
w.r.t XLA failure , yea please feel free to send a PR to pytorch/xla. We probably need to land these two PRs approx. same time.

**Pull request closed without merging**